### PR TITLE
PDF export dialog and info screen (EXPOSUREAPP-9187)

### PR DESCRIPTION
**Pull request by AlexanderAlferov**
Dialog and info screen for certificate export to PDF.

Screenshot 2021-08-26 at 18 42 18

How to test:
1) Scan Vaccination, Test and Recovery certificates.
2) In `VaccinationDetailsViewModel`, `TestCertificateDetailsViewModel` and `RecoveryCertificateDetailsViewModel` check `onExport` function.
3) Change `exportPossible` to `false` to show dialog, to `true` to continue navigation to info screen.

Actual navigation parameters and pdf error handling would be covered in next PRs.

### CODE REVIEW

**Review by InkenTownsend — APPROVED** [review]
UA approved

**Review by jurajkusnier — COMMENTED** [review]
Looks good, just few notes

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/ui/details/VaccinationDetailsViewModel.kt [hunk 1]
```diff
@@ -40,6 +40,8 @@ class VaccinationDetailsViewModel @AssistedInject constructor(
     val errors = SingleLiveEvent<Throwable>()
     val events = SingleLiveEvent<VaccinationDetailsNavigation>()
 
+    val exportError = SingleLiveEvent<Unit>()
```
[the remark this answers is not in the document]
Comment (AlexanderAlferov) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/ui/details/VaccinationDetailsViewModel.kt:44]: I would like to set it as separate SingleLiveEvent, as logically it is not the same as errors.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/recovery/ui/details/RecoveryCertificateDetailsViewModel.kt [hunk 2]
```diff
@@ -58,6 +62,14 @@ class RecoveryCertificateDetailsViewModel @AssistedInject constructor(
         recoveryCertificateRepository.acknowledgeState(containerId)
     }
 
+    fun onExport() {
+        if (recoveryCertificate.value?.certificateCountry != DccCountry("DE").displayName()) {
```
Comment (jurajkusnier) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/recovery/ui/details/RecoveryCertificateDetailsViewModel.kt:66]: What about something like
```kotlin
fun CwaCovidCertificate.canBeExported() = certificateCountry == DccCountry("DE").displayName() 
```
to don't repeat the implementation in all ViewModels?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/ui/details/VaccinationDetailsFragment.kt [hunk 3]
```diff
@@ -190,4 +202,15 @@ class VaccinationDetailsFragment : Fragment(R.layout.fragment_vaccination_detail
             }
         }.show()
     }
+
+    private fun showExportErrorDialog() {
```
Comment (jurajkusnier) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/ui/details/VaccinationDetailsFragment.kt:206]: Can you extract the dialog implementation? You are using exactly the same code 3 times

**Review by jurajkusnier — APPROVED** [review]

**Pull request merged**